sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosChangeFeedRequestOptions.java

```diff
@@ -432,4 +434,49 @@ public CosmosChangeFeedRequestOptions setThroughputControlGroupName(String throu
         this.throughputControlGroupName = throughputControlGroupName;
         return this;
     }
+
+    /**
+     * Sets the custom change feed request option value by key
+     *
+     * @param name  a string representing the custom option's name
+     * @param value a string representing the custom option's value
+     *
+     * @return the CosmosChangeFeedRequestOptions.
+     */
+    CosmosChangeFeedRequestOptions setHeader(String name, String value) {
+        if (this.customOptions == null) {
+            this.customOptions = new HashMap<>();
+        }
+        this.customOptions.put(name, value);
+        return this;
+    }
+
+    /**
+     * Gets the custom change feed request options
+     *
+     * @return Map of custom request options
+     */
+    Map<String, String> getHeaders() {
+        return this.customOptions;
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // the following helper/accessor only helps to access this class outside of this package.//
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    static {
+        ImplementationBridgeHelpers.CosmosChangeFeedRequestOptionsHelper.setCosmosChangeFeedRequestOptionsAccessor(
+            new ImplementationBridgeHelpers.CosmosChangeFeedRequestOptionsHelper.CosmosChangeFeedRequestOptionsAccessor() {
+
+                @Override
+                public CosmosChangeFeedRequestOptions setHeader(CosmosChangeFeedRequestOptions changeFeedRequestOptions, String name, String value) {
```

Comment:
I don't understand why is this needed? can we get away from it by toggling some boolean in the ChangeFeedProcessorBuilderImpl or something like that depending on encryption or not cases? This is only needed for a set of specific values that need to be added and not some arbitrary ones, right?

Reply:
This is for to pass encryption related header in cf request , this is line with item and query options. This is non public method so we shoudl be fine

Reply:
Discussed offline, this is fine